pyomo/contrib/doe/examples/reactor_compute_factorial_FIM.py

```diff
@@ -0,0 +1,98 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2024
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+from pyomo.common.dependencies import numpy as np
+
+from pyomo.contrib.doe.examples.reactor_experiment import ReactorExperiment
+from pyomo.contrib.doe import DesignOfExperiments
+
+import pyomo.environ as pyo
+
+import json
+from pathlib import Path
+
+
+# Example to run a DOE on the reactor
+def run_reactor_doe():
+    # Read in file
+    DATA_DIR = Path(__file__).parent
+    file_path = DATA_DIR / "result.json"
+
+    f = open(file_path)
+    data_ex = json.load(f)
```

Comment:
You should use a context manager when working with files to ensure that the file always gets closed